lib/poller.js

```diff
@@ -33,28 +33,54 @@ class Poller extends EventEmitter {
     } catch (err) {
       this.emit('error', err)
     }
-    Object.assign(this.cache, { serializedData: this.generateSerializedData() })
-    this.updateSerializedDataSubsets()
+    Object.assign(this.cache, { serializedData: await this.generateSerializedData() })
+    await this.updateSerializedDataSubsets()
   }
 
-  updateSerializedDataSubsets () {
-    Object.keys(this.cache.serializedDataSubsets).forEach(key => {
-      this.cache.serializedDataSubsets[key] = this.generateSerializedData(key.split('.'))
+  async updateSerializedDataSubsets () {
+    Object.keys(this.cache.serializedDataSubsets).forEach(async key => {
+      this.cache.serializedDataSubsets[key] = await this.generateSerializedData(key.split('.'))
     })
   }
 
-  getSerializedDataSubset (splits) {
+  async getSerializedDataSubset (splits) {
     const key = splits.sort().join('.')
     if (this.cache.serializedDataSubsets[key]) {
       return this.cache.serializedDataSubsets[key]
     }
-    const serializedData = this.generateSerializedData(splits)
+    const serializedData = await this.generateSerializedData(splits)
     this.cache.serializedDataSubsets[key] = serializedData
     return serializedData
   }
 
-  generateSerializedData (splits = []) {
+  async generateSerializedData (splits = []) {
     const latestCache = this.cache
+
+    // If generating serialized data for a subset of splits, get updated segments-related data for the subset
+    const serializingSegmentsForSubset = this.serializeSegments && splits.length
```

Comment:
this function is pretty long now, maybe create a `serializeSegmentsForSubset` helper function